app/fakefish.py

```diff
@@ -172,21 +164,36 @@ def set_env_vars(bmc_endpoint, username, password):
     my_env["BMC_PASSWORD"] = password
     return my_env
 
-def run():
+def run(port, debug, tls_mode):
     """
 
     """
-    app.run(host='0.0.0.0', port=port, debug=debug, ssl_context='adhoc')
+    if tls_mode == 'adhoc':
+        app.run(host='0.0.0.0', port=port, debug=debug, ssl_context='adhoc')
+    elif tls_mode == 'disabled':
+        app.run(host='0.0.0.0', port=port, debug=debug)
+    else:
+        if os.path.exists('./cert.pem') and os.path.exists('./cert.key'):
+            app.run(host='0.0.0.0', port=port, debug=debug, ssl_context=('./cert.pem', './cert.key'))
+        else:
+            app.logger.error('cert.pem or cert.key not found.')
+            exit()
 
 
 if __name__ == '__main__':
-
-    bmc_ip = os.environ.get('BMC_IP', None)
-    if bmc_ip is None:
-        app.logger.error('Configure the BMC IP using the environment variable BMC_IP')
-        exit()
+    parser = argparse.ArgumentParser(description='FakeFish, an experimental RedFish proxy that calls shell scripts for executing hardware actions.')
+    parser.add_argument('--tls-mode', type=str, choices=['adhoc', 'self-signed', 'disabled'], default='adhoc', help='Configures TLS mode. \'self-signed\' mode expects cert.pem and cert.key files within the app path. (default: %(default)s)')
```

Comment:
Could you please keep the code wrapped at column 80?

Reply:
Done.